compiler/hash-typecheck/src/ops/simplify.rs

```diff
@@ -414,11 +453,15 @@ impl<'tc> Simplifier<'tc> {
             Level2Term::Trt(trt_def_id) => {
                 does_not_support_prop_access(access_term)?;
 
+                let Field::Named(name) = access_term.name else {
+                    unreachable!()
+                };
```

Comment:
Same as above

Reply:
It's handled in `does_not_support_prop_access` in the call above.